src/hatch_openzim/files_install.py

```diff
@@ -0,0 +1,199 @@
+        _process_get_file_action(
+            base_target_dir=base_target_dir, source=source, action_data=action_data
+        )
+
+    elif action == "extract_all":
+        _process_extract_all_action(
+            base_target_dir=base_target_dir, source=source, action_data=action_data
+        )
+
+    elif action == "extract_items":
+        _process_extract_items_action(
+            base_target_dir=base_target_dir, source=source, action_data=action_data
+        )
+
+    else:
+        raise Exception(f"Unsupported action '{action}'")
+
+    log.info("    Done")
+
+
+def _process_get_file_action(
+    base_target_dir: Path, source: str, action_data: Dict[str, Any]
+):
+    """downloads one file to a given location"""
+    target_file = action_data.get("target_file", None)
+    if not target_file:
+        raise Exception("target_file is mandatory when action='get_file'")
+    target_dir = action_data.get("target_dir", None)
+    local_dir = base_target_dir
+    if target_dir:
+        local_dir = base_target_dir / str(target_dir)
+        local_dir.mkdir(parents=True, exist_ok=True)
+    local_file = local_dir / str(target_file)
+    local_file.unlink(missing_ok=True)
+    _download_file(source, local_file)
+
+
+def _process_extract_all_action(
+    base_target_dir: Path, source: str, action_data: Dict[str, Any]
+):
+    """extacts all Zip content to a given location
+
+    Supports the remove attribute which allows to delete some Zip content
+    """
+    target_dir = action_data.get("target_dir", None)
+    if not target_dir:
+        raise Exception("target_dir is mandatory when action='extract_all'")
+    target_dir = base_target_dir / str(target_dir)
+    if target_dir.exists():
+        shutil.rmtree(target_dir)
+    if not target_dir.parent.exists():
+        target_dir.parent.mkdir(parents=True, exist_ok=True)
+    _extract_zip_from_url(url=source, extract_to=target_dir)
+    if "remove" in action_data:
+        _remove_items(globs=action_data["remove"], directory=target_dir)
+
+
+def _process_extract_items_action(
+    base_target_dir: Path, source: str, action_data: Dict[str, Any]
+):
+    """extacts some items of a content to some given locations
+
+    Supports the remove attribute which allows to delete some items content after
+    extraction
+    """
+    zip_paths = action_data.get("zip_paths", None)
+    if not zip_paths:
+        raise Exception("zip_paths is mandatory when action='extract_items'")
+    target_paths = action_data.get("target_paths", None)
+    if not target_paths:
+        raise Exception("target_paths is mandatory when action='extract_items'")
+    if len(zip_paths) != len(target_paths):
+        raise Exception(
+            f"zip_paths and target_paths must have same length ({len(zip_paths)} !="
+            f" {len(target_paths)})"
+        )
+
+    with tempfile.TemporaryDirectory() as tempdir:
+        _extract_zip_from_url(url=source, extract_to=tempdir)
+        for i in range(0, len(zip_paths)):
```

Comment:
No `i` 😉 ; `index` is better. You could even have `for index, item_src in enumerate(zip_paths):`